app/services/trading_strategies.py

```
@@ -4153,6 +4153,405 @@ def _calculate_portfolio_sortino(self, positions: List[Dict]) -> float:
+                return {
+                    "success": True, 
+                    "price": fallback_price,
+                    "symbol": symbol,
+                    "fallback": True,
+                    "timestamp": datetime.utcnow().isoformat()
+                }
+        except Exception as e:
+            self.logger.error(f"Price fetch failed for {symbol}: {e}")
+            return {"success": False, "error": str(e)}
+
+    async def _get_perpetual_funding_info(self, symbol: str, exchange: str) -> Dict[str, Any]:
```

Ruff (0.12.2)

Unused method argument: `exchange`

(ARG002)